lib/matplotlib/image.py

```diff
@@ -1004,106 +904,30 @@ def __init__(self, fig,
 
         kwargs are an optional list of Artist keyword args
         """
-        martist.Artist.__init__(self)
-        cm.ScalarMappable.__init__(self, norm, cmap)
-        if origin is None:
-            origin = rcParams['image.origin']
-        self.origin = origin
+        _ImageBase.__init__(
+            self, None, norm=norm, cmap=cmap, origin=origin)
         self.figure = fig
         self.ox = offsetx
         self.oy = offsety
         self.update(kwargs)
         self.magnification = 1.0
 
-    def contains(self, mouseevent):
-        """Test whether the mouse event occured within the image."""
-        if six.callable(self._contains):
-            return self._contains(self, mouseevent)
-        xmin, xmax, ymin, ymax = self.get_extent()
-        xdata, ydata = mouseevent.x, mouseevent.y
-
-        if xdata is not None and ydata is not None:
-            inside = ((xdata >= xmin) and (xdata <= xmax) and
-                      (ydata >= ymin) and (ydata <= ymax))
-        else:
-            inside = False
-
-        return inside, {}
-
-    def get_size(self):
-        """Get the numrows, numcols of the input image"""
-        if self._A is None:
-            raise RuntimeError('You must first set the image array')
-
-        return self._A.shape[:2]
-
     def get_extent(self):
         """Get the image extent: left, right, bottom, top"""
         numrows, numcols = self.get_size()
-        return (-0.5+self.ox, numcols-0.5+self.ox,
-                -0.5+self.oy, numrows-0.5+self.oy)
+        return (-0.5 + self.ox, numcols-0.5 + self.ox,
+                -0.5 + self.oy, numrows-0.5 + self.oy)
 
-    def set_data(self, A):
-        """Set the image array."""
-        cm.ScalarMappable.set_array(self, cbook.safe_masked_invalid(A))
-        self.stale = True
+    def make_image(self, renderer, magnification=1.0, unsampled=False):
+        bbox = Bbox([[self.ox, self.oy],
+                     [self.ox + self._A.shape[1], self.oy + self._A.shape[0]]])
+        clip = Bbox([[0, 0], [renderer.width, renderer.height]])
+        return self._make_image(
+            self._A, bbox, bbox, clip, magnification=magnification,
+            unsampled=unsampled, round_to_pixel_border=False)
 
-    def set_array(self, A):
-        """Deprecated; use set_data for consistency with other image types."""
-        self.set_data(A)
-
-    def make_image(self, magnification=1.0):
-        if self._A is None:
-            raise RuntimeError('You must first set the image array')
-
-        A = self._A
-        if self.origin == 'upper':
-            A = A[::-1]
-
-        x = self.to_rgba(A, bytes=True)
-        self.magnification = magnification
```

Comment:
Is `self.magnification` still a thing after this rewrite? I don't think I saw it anywhere. Hopefully, that isn't an issue.